File: AI.Core/OperatorsImplimentation/Operators.cs
using System.Collections.Generic;

namespace AI.Core.OperatorsImplimentation
{
    /// <summary>
    /// Dictionary collection of Entity Operators
    /// </summary>
    public class Operators : Dictionary<ulong, Operator>
    {
        private ulong _currentId;

        /// <summary>
        /// Returns new unique Operator Id that is used for Operator creation and for Operator adding to Operators dictionary
        /// </summary>
        public ulong GetNewUniqueId()
        {
            do
            {
                _currentId++;
            } while (ContainsKey(_currentId));

            return _currentId;
        }
    }
}

File: NativeCore/Operators.cpp
#include "Operators.h"
#include "stdafx.h"


//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System::Collections::Generic;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{

			unsigned long long Operators::GetNewUniqueId()
			{
				do
				{
					_currentId++;
				} while (ContainsKey(_currentId));

				return _currentId;
			}

			void Operators::InitializeInstanceFields()
			{
				_currentId = 0;
			}
		}
	}
}
